test/demo/createCenter.inttest.ts

```diff
@@ -0,0 +1,203 @@
+            createPromise = notifsRemote.create(testWindow.identity, options);
+
+            // We want to be sure the operation is completed, but don't care if it succeeds
+            await createPromise.catch(() => {});
+        });
+
+        afterEach(async () => {
+            // Cleanup the created notification (just in case it does get made)
+            await notifsRemote.clear(testWindow.identity, options.id!);
+        });
+
+        test('The promise rejects with a suitable error message', async () => {
+            await expect(createPromise).rejects.toThrow(/Invalid arguments passed to createNotification/);
+        });
+
+        test('A card is not added to the notification center', async () => {
+            const noteCards = await getCardsByNotification(testApp.identity.uuid, options.id!);
+            expect(noteCards).toEqual([]);
+        });
+
+        test('The notification is not added to the persistence store', async () => {
+            const storedNotes = await getAppNotifications(testApp.identity.uuid);
+
+            expect(storedNotes).toEqual([]);
+        });
+
+        test('The notification does not appear in the result of getAll', async () => {
+            const appNotes = await notifsRemote.getAll(testWindow.identity);
+            expect(appNotes).toEqual([]);
+        });
+    });
+
+    describe('When options does not include id', () => {
+        const options: NotificationOptions = {body: 'Test Notification Body', title: 'Test Notificaiton Title'};
+
+        let createPromise: Promise<Notification>;
+        let note: Notification;
+        beforeEach(async () => {
+            createPromise = notifsRemote.create(testWindow.identity, options);
+
+            // We want to be sure the operation is completed, but don't care if it succeeds
+            note = await createPromise.catch(() => ({} as Notification));
+        });
+
+        test('The notification is created as expected', async () => {
+            await expect(createPromise).resolves;
+            expect(note).toMatchObject(options);
+
+            // Card is created
+            const noteCards = await getCardsByNotification(testApp.identity.uuid, note.id);
+            expect(noteCards).toHaveLength(1);
+
+            // Card is correct
+            await assertDOMMatches(testApp.identity.uuid, note);
+
+            // Notification is persisted
+            await assertNotificationStored(testWindow.identity, note);
+        });
+
+        test('An id is generated by the service and returned in the hydrated object', async () => {
+            await expect(createPromise).resolves;
+
+            expect(note).toMatchObject(options);
+            expect(note.id).toMatch(/[0-9]{9}/); // Random 9-digit numeric string
+        });
+    });
+
+    describe.each([1, 2, 3])('With %i button(s)', numButtons => {
+        const options: NotificationOptions = {body: 'Test Notification Body', title: 'Test Notificaiton Title', buttons: []};
```

Comment:
Thinking out loud:
I feel like as we add more components to notifications options this will become more bloated with testing all the different possibilities. Especially since the same component can be rendered in 2 different places (center/toast) leading to duplicated tests. We should probably start using `enzyme` to unit test our react stuff, then we can just have integration tests interact with those components to test the expected outcomes.

Reply:
Its a fair point. The goal here is just a basic level of end-to-end stuff, i.e. if I make this call, does this thing happen. I don't think I would ever consider doing a full test of every conceivable option as part of this suite, but a general sample of some different combinations I see as useful, especially as we don't have any component testing currently in place.